docs(cketh/ckerc20): update docs (#2992)
dfinity/ic: rs/ethereum/cketh/docs/cketh.adoc

== Deposit: ETH to ckETH

----
 ┌────┐                               ┌───────────────┐                                                ┌──────┐
 │User│                               │Helper Contract│                                                │Minter│
 └─┬──┘                               └───────┬───────┘                                                └──┬───┘
   │                                          │                                                           │
   │depositEth(amount, principal, subaccount) │                                                           │
   │─────────────────────────────────────────>│                                                           │
   │                                          │                                                           │
   │                                          │       get_events                                          │
   │                                          │<──────────────────────────────────────────────────────────│
   │                                          │                                                           │

>    │                                          │ReceivedEthOrErc20(0x, user, amount, principal, subaccount)│

Reviewer: Potentially surround 0x with quotes to emphasize it is a fixed value as opposed to the other parameters.

Author: make sense, done.

>    │                                          │ReceivedEthOrErc20(0x, user, amount, principal, subaccount)│

Reviewer: ```suggestion
   │                                          │ReceivedEthOrErc20(`0x`, user, amount, principal, subaccount)│
```

.Address of the helper smart contract
====
The address of the helper smart contract may change in the future when the minter is upgraded.

> Please verify the address of the helper contract **before** any important transfer by checking the address specified in the field `deposit_with_subaccount_helper_contract_address` in the response of `get_minter_info`, which can be queried as follows

Reviewer: ```suggestion
Please verify the address of the helper contract **before** any important transfer by comparing it to the address specified in the field `deposit_with_subaccount_helper_contract_address` in the response of `get_minter_info`, which can be queried as follows
```

----
====

Call the link:https://etherscan.io/address/0x18901044688D3756C35Ed2b36D93e6a5B8e00E68#writeContract[minter helper contract] `depositEth` method of the helper smart contract (can be called from the contract's Etherscan page) specifying

. The amount of ETH being deposited.
. The IC principal encoded as a `bytes32`.

> . The IC ledger subaccount (use `0x` for the default subaccount), which is a 32-byte array, encoded as an hexadecimal string and prefixed by `0x`. The IC principal together with the IC ledger subaccount uniquely identify an IC ledger account to which the amount of ckETH should be minted.

Reviewer: ```suggestion
. The IC ledger subaccount (use `0x` for the default subaccount), which is a 32-byte array, encoded as a hexadecimal string and prefixed by `0x`. The IC principal together with the IC ledger subaccount uniquely identify an IC ledger account to which the amount of ckETH should be minted.
```
